RadixWallet/Clients/SubmitTransactionClient/SubmitTransactionClient+Live.swift

```diff
@@ -5,79 +5,42 @@ extension SubmitTransactionClient: DependencyKey {
 	public static let liveValue: Self = {
 		@Dependency(\.gatewayAPIClient) var gatewayAPIClient
 
-		let transactionStatusUpdates: TransactionStatusUpdates = { txID, pollStrategy in
+		let hasTXBeenCommittedSuccessfully: HasTXBeenCommittedSuccessfully = { txID in
 			@Dependency(\.continuousClock) var clock
 
-			let statusSubject = AsyncCurrentValueSubject<TransactionStatusUpdate>(.init(txID: txID, result: .success(.pending)))
-
-			@Sendable func pollTransactionStatus() async throws -> GatewayAPI.TransactionStatus {
+			@Sendable func pollTransactionStatus() async throws -> GatewayAPI.TransactionStatusResponse {
 				let txStatusRequest = GatewayAPI.TransactionStatusRequest(
 					intentHash: txID.asStr()
 				)
 				let txStatusResponse = try await gatewayAPIClient.transactionStatus(txStatusRequest)
-				return txStatusResponse.status
+				return txStatusResponse
 			}
 
-			let pollCountHolder = ActorIsolated<Int>(0)
-
-			Task {
-				while !statusSubject.value.result.isComplete {
-					await pollCountHolder.withValue { pollCount in
-						if pollCount >= pollStrategy.maxPollTries {
-							statusSubject.send(.init(
-								txID: txID,
-								result: .failure(.failedToGetTransactionStatus(
-									txID: txID,
-									error: .init(pollAttempts: pollCount)
-								))
-							))
-						} else {
-							pollCount += 1
-						}
-					}
-					try? await clock.sleep(for: .seconds(pollStrategy.sleepDuration))
-					do {
-						let status = try await pollTransactionStatus()
-						statusSubject.send(.init(txID: txID, result: .success(status)))
-					} catch {
-						loggerGlobal.error("Failed to poll transaction status: \(error.localizedDescription)")
-						switch error {
-						case is BadHTTPResponseCode, is BadHTTPResponseCode, is ResponseDecodingError:
-							/// Terminate polling, GW returned wrong response
-							await statusSubject.send(.init(
-								txID: txID,
-								result: .failure(.failedToGetTransactionStatus(txID: txID, error: .init(pollAttempts: pollCountHolder.value)))
-							))
-						default:
-							/// Continue polling, the poll failed due to internal URLSession error.
-							break
-						}
-					}
+			while true {
```

Comment:
can be made into a function we recursively call and add 0.5 or 1 sec after each attempt, as to not load GW to much.

Reply:
we could do that, but this pretty much keeps the same behaviour as we currently have in production. Will ask the network team about the load on the transaction status endpoint, I assume it is quite lightweight.

Reply:
Added 1 second increment.
Did keep the `while` loop instead of recursion, as to not grow the call stack.